src/main/java/com/innoq/spring/cookie/flash/CookieFlashMapManager.java

```diff
@@ -87,16 +97,14 @@ protected Object getFlashMapsMutex(HttpServletRequest request) {
     private List<FlashMap> decode(String value) {
         final String[] signatureAndPayload = reverse(value).split("--", 2);
         if (signatureAndPayload.length != 2) {
-            // TODO logging
-            return null;
+            return verificationFailureHandler.onInvalidValue(value);
```

Comment:
What is the semantics of this failure, i.e. what went wrong? And is is worth distinguishing between `onInvalidValue()` and `onInvalidSignature()`? In both cases, the content of the cookie was not as expected.